sequencer/src/block/queryable.rs

```diff
@@ -327,26 +345,29 @@ impl TxInclusionProof {
 
         // Verify proof for tx table entries.
         // Start index missing for the 0th tx
-        let index: usize = tx_index.tx_idx; // TODO fix in https://github.com/EspressoSystems/espresso-sequencer/issues/1010
+        let index: usize = tx_index.tx_idx;
         let mut tx_table_range_bytes =
             Vec::with_capacity(2usize.checked_mul(TxTableEntry::byte_len())?);
         let start = if let Some(tx_table_range_start) = &self.tx_table_range_start {
             if index == 0 {
                 return None; // error: first tx should have empty start index
             }
             tx_table_range_bytes.extend(tx_table_range_start.to_bytes());
-            index * TxTableEntry::byte_len()
+            index
+                .checked_mul(TxTableEntry::byte_len())?
+                .checked_add(self.ns_range.start)?
         } else {
             if index != 0 {
                 return None; // error: only the first tx should have empty start index
             }
-            TxTableEntry::byte_len()
+            TxTableEntry::byte_len().checked_add(self.ns_range.start)?
         };
         tx_table_range_bytes.extend(self.tx_table_range_end.to_bytes());
         let range = start
             ..index
                 .checked_add(2)?
-                .checked_mul(TxTableEntry::byte_len())?;
+                .checked_mul(TxTableEntry::byte_len())?
+                .checked_add(self.ns_range.start)?;
```

Comment:
It looks like we compute the same offsets and ranges in multiple places. Wondering if it would be worth it to add some helper methods for it and briefly comment those. I think it would make it easier to read the code, at least for people like me who aren't familiar with this part of the code.